code/modules/mob/living/simple_animal/bot/vibebot.dm

```diff
@@ -55,3 +48,60 @@
 		start_patrol()
 	if(mode == BOT_PATROL)
 		bot_patrol()
+
+/mob/living/simple_animal/bot/vibebot/turn_off()
+	remove_atom_colour(TEMPORARY_COLOUR_PRIORITY)
+	. = ..()
+
+/mob/living/simple_animal/bot/vibebot/proc/remove_colors()
+	remove_atom_colour(TEMPORARY_COLOUR_PRIORITY)
+	set_light_color(null)
+
+/mob/living/simple_animal/bot/vibebot/proc/vibe()
+	remove_atom_colour(TEMPORARY_COLOUR_PRIORITY)
+	add_atom_colour("#[random_color()]", TEMPORARY_COLOUR_PRIORITY)
+	set_light_color(color)
```

Comment:
I didn't actually consider it until now, buy you could put all of this on the `datum/action` itself instead of on the bot. Would be kinda cool, I think.

Reply:
i hate you + L + ok sure